dataprofiler/data_readers/filepath_or_buffer.py

```diff
@@ -79,15 +80,17 @@ def __enter__(self):
 
         return self._filepath_or_buffer
 
-    def __exit__(self, exc_type, exc_value, exc_traceback):
+    def __exit__(self, exc_type: Any, exc_value: Any, exc_traceback: Any) -> None:
         """Release resources."""
         # Need to detach buffer if wrapped (i.e. BytesIO opened with 'r')
         if self._is_wrapped:
+            self._filepath_or_buffer = cast(TextIOWrapper, self._filepath_or_buffer) # guaranteed by self._is_wrapped
```

Comment:
potential for follow-up around casting and ensuring typing declaration is right and/or code is corrected to avoid the change of a wrong type